gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -737,7 +737,7 @@ private int peekNumber() throws IOException {
     }
 
     // We've read a complete number. Decide if it's a PEEKED_LONG or a PEEKED_NUMBER.
-    if (last == NUMBER_CHAR_DIGIT && fitsInLong && (value != Long.MIN_VALUE || negative) && (value!=0 || false==negative)) {
+    if (last == NUMBER_CHAR_DIGIT && fitsInLong && (value != Long.MIN_VALUE || negative) && (value!=0 || !negative)) {
```

Comment:
In a different pull request there was a discussion about further simplifying this and also adding comments explaining these checks: https://github.com/google/gson/pull/2221#discussion_r993964588

Would you mind implementing that? It is not clear if the author of that pull request intends to update the pull request.